base_layer/core/src/consensus/consensus_manager.rs

```diff
@@ -271,3 +317,64 @@ pub enum ConsensusBuilderError {
     #[error("Cannot set a genesis block with a network other than LocalNet")]
     CannotSetGenesisBlock,
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::consensus::ConsensusConstantsBuilder;
+
+    #[test]
+    fn test_epoch_to_height_and_back() {
+        let manager = ConsensusManager::builder(Network::LocalNet)
+            .add_consensus_constants(
+                ConsensusConstantsBuilder::new(Network::LocalNet)
+                    .with_effective_height(0)
+                    .with_vn_epoch_length(15)
+                    .build(),
+            )
+            .add_consensus_constants(
+                ConsensusConstantsBuilder::new(Network::LocalNet)
+                    .with_effective_height(100)
+                    .with_vn_epoch_length(6)
+                    .build(),
+            )
+            .add_consensus_constants(
+                ConsensusConstantsBuilder::new(Network::LocalNet)
+                    .with_effective_height(200)
+                    .with_vn_epoch_length(8)
+                    .build(),
+            )
+            .add_consensus_constants(
+                ConsensusConstantsBuilder::new(Network::LocalNet)
+                    .with_effective_height(300)
+                    .with_vn_epoch_length(13)
+                    .build(),
+            )
+            .add_consensus_constants(
+                ConsensusConstantsBuilder::new(Network::LocalNet)
+                    .with_effective_height(400)
+                    .with_vn_epoch_length(17)
+                    .build(),
+            )
+            .add_consensus_constants(
+                ConsensusConstantsBuilder::new(Network::LocalNet)
+                    .with_effective_height(500)
+                    .with_vn_epoch_length(7)
+                    .build(),
+            )
+            .build()
+            .unwrap();
+        assert_eq!(manager.block_height_to_epoch(99), VnEpoch(6)); // The next epoch should change at 105
+        assert_eq!(manager.block_height_to_epoch(100), VnEpoch(7)); // But with the new length the epoch should change right away
+        assert_eq!(manager.block_height_to_epoch(199), VnEpoch(23)); // The next epoch should change at 202
+        assert_eq!(manager.block_height_to_epoch(202), VnEpoch(23)); // But we have new length with size +2 so the epoch change will happen at 204
+        assert_eq!(manager.block_height_to_epoch(204), VnEpoch(24));
+        // Now test couple more back and forth
+        for epoch in 0..=100 {
+            assert_eq!(
+                manager.block_height_to_epoch(manager.epoch_to_block_height(VnEpoch(epoch))),
+                VnEpoch(epoch)
+            );
+        }
+    }
```

Comment:
Maybe we can be more explicit in this test
```suggestion
        let mut epoch = manager.block_height_to_epoch(0).as_u64();
        for height in 0..600 {
            assert!(manager.block_height_to_epoch(height).as_u64() >= epoch);
            epoch = manager.block_height_to_epoch(height).as_u64();
        }
    }
```

Reply:
I've created a new for the non decreasing `block_height_to_epoch`.